src/core/Song.cpp

```diff
@@ -121,6 +121,7 @@ Song::Song() :
 
 	for (auto& scale : m_scales) {scale = std::make_shared<Scale>();}
 	for (auto& keymap : m_keymaps) {keymap = std::make_shared<Keymap>();}
+	for (auto& timeline : m_timelines) { connect(this, &Song::bufferProcessed, &timeline, &Timeline::updatePosition, Qt::QueuedConnection); }
```

Comment:
I agree, I think the `bufferProcessed` signal and `updatePosition` slot can probably be removed.

Except that, when I remove them and instead rely on `setTicks(ticks)` to emit the signal, the `processNextBuffer` function in `Song` does not update the playhead position, so it stays still during playback. I believe this may be due to `processNextBuffer` being on a different thread?

But then it somehow fixes itself when I specify `Qt::DirectConnection` in all the `connect()`s. I'm not sure why.